File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode invertTree(TreeNode root) {
        doInvertTree(root);
        return root;
    }

    private void doInvertTree(TreeNode node) {
        if (node == null) {
            return;
        }
        var left = node.left;
        var right = node.right;
        node.left = right;
        node.right = left;
        doInvertTree(left);
        doInvertTree(right);
    }
}

File: Data Structures & Algorithms/serialize-and-deserialize-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Codec:
    
    # Encodes a tree to a single string.
    def serialize(self, root: Optional[TreeNode]) -> str:
        if not root:
            return ""
        encoded = ""
        queue = deque()
        queue.append(root)
        while queue:
            node = queue.popleft()
            if node:
                queue.append(node.left)
                queue.append(node.right)
                encoded += f"{node.val},"
            else:
                encoded += "None,"
        return encoded[:-1]


        
    # Decodes your encoded data to tree.
    def deserialize(self, data: str) -> Optional[TreeNode]:
        if data == "":
            return None
        values = deque(data.split(','))
        root = TreeNode(int(values.popleft()))
        queue = deque()
        queue.append(root)
        while values:
            node = queue.popleft()
            left = values.popleft()
            right = values.popleft()
            if left != 'None':
                node.left = TreeNode(int(left))
                queue.append(node.left)
            if right != 'None':
                node.right = TreeNode(int(right))
                queue.append(node.right)
        return root
